File: api/src/main/java/com/aistore/module/spu/entity/Spu.java
package com.aistore.module.spu.entity;

import com.baomidou.mybatisplus.annotation.FieldFill;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableLogic;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

/** 标准产品单元实体（表 spu，品类层） */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@TableName("spu")
public class Spu {

    @TableId(value = "id", type = IdType.AUTO)
    private Long id;
    private String spuCode;
    private String spuName;
    private String categoryCode;
    private String brand;
    private String baseUnit;
    private Integer status;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
    private Long createdBy;
    private Long updatedBy;

    @TableLogic
    private Integer deleted;
}

File: web/src/types/spu.ts
/* ========================================
   SPU 模块类型（对齐 OpenAPI 契约）
   ======================================== */

/** SPU 详情（对齐 Spu schema） */
export interface Spu {
  id: number;
  spuCode: string;
  spuName: string;
  categoryCode: string;
  categoryName?: string | null;
  brand?: string | null;
  baseUnit: string;
  status: 0 | 1;
  createdAt: string;
  updatedAt: string;
  createdBy?: number | null;
  updatedBy?: number | null;
}

/** SPU 列表项（对齐 SpuSummary schema） */
export interface SpuSummary {
  id: number;
  spuCode: string;
  spuName: string;
  categoryCode: string;
  categoryName?: string | null;
  baseUnit: string;
  status: 0 | 1;
  createdAt: string;
}

/** SPU 列表分页响应 */
export interface SpuListResponse {
  items: SpuSummary[];
  total: number;
  page: number;
  pageSize: number;
  totalPages: number;
}

/** 创建 SPU 请求 */
export interface CreateSpuRequest {
  spuCode: string;
  spuName: string;
  categoryCode: string;
  brand?: string;
  baseUnit: string;
  status?: 0 | 1;
}

/** 更新 SPU 请求（spu_code 不可改，状态走独立接口） */
export interface UpdateSpuRequest {
  spuName?: string;
  categoryCode?: string;
  brand?: string;
  baseUnit?: string;
}

/** SPU 查询参数 */
export interface SpuQueryParams {
  keyword?: string;
  categoryCode?: string;
  status?: 0 | 1;
  page?: number;
  pageSize?: number;
}
